### PR TITLE
[tests-only] Adds e2e test for changing public link role

### PR DESCRIPTION
**Pull request by SwikritiT**
## Description

 Adds step for: `Alice changes role to editor, change expiration date`

## Related Issue

- Part of: https://github.com/owncloud/web/issues/6389

## How Has This Been Tested?

- local
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [x] e2e tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/23424/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23424/tracing/alice-shares-folder-with-file-to-brian-alice-2022-3-9-09-51-55.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23424/tracing/alice-shares-folder-with-file-to-brian-brian-2022-3-9-09-52-08.zip 


</p></details>

**Comment by SwikritiT** [comment]
This PR needs to be merged first https://github.com/owncloud/web/pull/6541

**Review by pascalwengerter — COMMENTED** [review]
CI failed with linting errors. Also please wait for @fschade's feedback (he's not in the office today) on E2E tests before merging

File: tests/e2e/cucumber/features/integrations/link.feature
Comment (fschade) [inline tests/e2e/cucumber/features/integrations/link.feature:81]: Please do not log in alice again, just keep her logged in the previous steps, then at the end log her out again.

Logging out is at the needed to tell playwright that the page context can be closed and de-registered.
The only reason to log-out and then re-login inside a test is when the steps between those take longer then the timeout from playwright.

File: tests/e2e/support/objects/app-files/link/actions.ts
Comment (fschade) [inline tests/e2e/support/objects/app-files/link/actions.ts:27]: the K at the end should be lower-case.
Please also use an object arg in the function which makes a future refactoring more easy.

File: tests/e2e/support/objects/app-files/link/index.ts
Comment (fschade) [inline tests/e2e/support/objects/app-files/link/index.ts:28]: why not just use `args.newName` here, then we can skip all the boilerplate above.

**Review by fschade — CHANGES_REQUESTED** [review]
please see comments, thanks for you contribution.

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/24392/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/24392/tracing/public-link-alice-2022-4-4-05-00-30.zip 


</p></details>

**Comment by fschade** [comment]
@SwikritiT can you rebase please

**Comment by pascalwengerter** [comment]
> @SwikritiT can you rebase please

Just merged #6541 so rebasing should be simpler (I hope 😬 )

**Comment by pascalwengerter** [comment]
Also, please use the "Squash and merge" option here (or squash commits locally while rebasing)

File: tests/e2e/support/objects/app-files/link/actions.ts
Comment (kiranparajuli589) [inline tests/e2e/support/objects/app-files/link/actions.ts:87]: use var

File: tests/e2e/support/objects/app-files/link/actions.ts
Comment (kiranparajuli589) [inline tests/e2e/support/objects/app-files/link/actions.ts:75]: use var

**Review by fschade — APPROVED** [review]
LGTM 🎸  Thanks for taking care

**Pull request merged**
